### PR TITLE
feat: add type hints for `rdflib.store` and `rdflib.plugins.stores`

### PR DESCRIPTION
**Pull request by aucampia**
# Summary of changes

Add type hints and aliases for `rdflib.store` and
`rdflib.plugins.stores` and also add a couple of more type hints and
aliases to `rdflib.graph`.

This PR contains no runtime changes.

Other changes:
- Changed some imports to be more specific (e.g. `import from
  rdflib.graph` instead of `import from rdflib`). This is to reduce
  the probability of circular imports.
- Ignore `E231` (missing whitespace after ',') in flake8 as black is
  managing the whitespaces and seems to be bumping heads with flake8
  with spaces after `,` sometimes.
- Install `berkeleydb-stubs` when doing extensive testing with tox.
- Added `devtools/diffrtpy.py` which is a script that can be used with
  `git difftool` to generate compact diffs for python code. This should
  make it a lot easier to review PRs that change type hints to verify
  that they don't have a runtime impact.

Compact diff: https://gist.github.com/aucampia/3f380070fbd8f0e8a99aea2df6cc1f06

# Checklist

- [x] Checked that there aren't other open pull requests for
  the same change.
- [x] Added tests for any changes that have a runtime impact.
- [x] Checked that all tests and type checking passes.
- For changes that have a potential impact on users of this project:
  - [x] Considered updating our changelog (`CHANGELOG.md`).
- [x] Considered granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork),
  so maintainers can fix minor issues and keep your PR up to date.

### CODE REVIEW

**Comment by aucampia** [comment]
I will publish this tomorrow, I'm writing a script to diff python source code with type hints and comments removed, so it is easier to review these PRs.

**Comment by aucampia** [comment]
Compact diff can be found here: https://gist.github.com/aucampia/3f380070fbd8f0e8a99aea2df6cc1f06

**Comment by aucampia** [comment]
@ajnelson-nist @mielvds tagging you in case you have time to review.

File: devtools/diffrtpy.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+This is to make it easier to review PRs that contain type hints.
+
+To use this script
+
+.. code-block:: bash
+    task run -- python -m pip install --upgrade strip-hints black python-minifier
+    PYLOGGING_LEVEL=INFO task run -- git difftool -y -x $(readlink -f devtools/diffrtpy.py) upstream/master | tee /var/tmp/compact.diff
+
+Then attatch ``/var/tmp/compact.diff`` to the PR.
```
Comment (ajnelson-nist) [inline devtools/diffrtpy.py:17]: Typo here.

Comment (aucampia) [reply devtools/diffrtpy.py]: Thanks, fixed now

**Review by gjhiggins — APPROVED** [review]
lgtm, will be v. useful for ADR

**Pull request merged**